connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/CloudEventHistoryBuilder.java

```diff
@@ -0,0 +1,32 @@
+package com.redhat.cloud.notifications.connector.email;
+
+import com.redhat.cloud.notifications.connector.http.HttpOutgoingCloudEventBuilder;
+import io.vertx.core.json.JsonObject;
+import jakarta.annotation.Priority;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Alternative;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import java.util.Optional;
+
+@ApplicationScoped
+@Alternative
+@Priority(0) // The value doesn't matter.
+public class CloudEventHistoryBuilder extends HttpOutgoingCloudEventBuilder {
+
+    public static final String TOTAL_RECIPIENTS_KEY = "total_recipients";
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        super.process(exchange);
+        int totalRecipients = Optional.ofNullable(exchange.getProperty(TOTAL_RECIPIENTS_KEY, Integer.class)).orElse(0);
```

Comment:
You can set a default value when getting a property in Camel.

```suggestion
        int totalRecipients = exchange.getProperty(TOTAL_RECIPIENTS_KEY, 0, Integer.class);
```